diff --git a/browser/components/urlbar/UrlbarUtils.sys.mjs b/browser/components/urlbar/UrlbarUtils.sys.mjs
index b97fa81cef276e4bb22414b7c182e00b4bb835dc..2c38f0ffcc03c65d5997feece481b1e8ccd3c702 100644
--- a/browser/components/urlbar/UrlbarUtils.sys.mjs
+++ b/browser/components/urlbar/UrlbarUtils.sys.mjs
@@ -81,6 +81,7 @@ export var UrlbarUtils = {
     RESTRICT_SEARCH_KEYWORD: "restrictSearchKeyword",
     SUGGESTED_INDEX: "suggestedIndex",
     TAIL_SUGGESTION: "tailSuggestion",
+    ZEN_ACTION: "zenAction",
   }),
 
   // Defines provider types.
@@ -140,6 +141,7 @@ export var UrlbarUtils = {
     OTHER_NETWORK: 6,
     ADDON: 7,
     ACTIONS: 8,
+    ZEN_ACTIONS: 9,
   }),
 
   // Per-result exposure telemetry.
@@ -271,6 +273,14 @@ export var UrlbarUtils = {
         telemetryLabel: "actions",
         uiLabel: "urlbar-searchmode-actions",
       },
+      {
+        source: this.RESULT_SOURCE.ZEN_ACTIONS,
+        restrict: lazy.UrlbarTokenizer.RESTRICT.WORKSPACE,
+        icon: "chrome://browser/skin/tabs.svg",
+        pref: "shortcuts.zen-actions",
+        telemetryLabel: "zen-actions",
+        uiLabel: "urlbar-searchmode-zen-actions",
+      },
     ];
   },
 
@@ -552,6 +562,8 @@ export var UrlbarUtils = {
           return this.RESULT_GROUP.HEURISTIC_FALLBACK;
         case "UrlbarProviderHistoryUrlHeuristic":
           return this.RESULT_GROUP.HEURISTIC_HISTORY_URL;
+        case "ZenUrlbarProviderGlobalActions":
+          return this.RESULT_GROUP.ZEN_ACTION;
         case "UrlbarProviderOmnibox":
           return this.RESULT_GROUP.HEURISTIC_OMNIBOX;
         case "UrlbarProviderRestrictKeywordsAutofill":
@@ -1329,6 +1341,9 @@ export var UrlbarUtils = {
         if (result.payload.keyword === lazy.UrlbarTokenizer.RESTRICT.ACTION) {
           return "restrict_keyword_actions";
         }
+        if (result.payload.keyword === lazy.UrlbarTokenizer.RESTRICT.) {
+          return "restrict_keyword_tabs";
+        }
     }
     return "unknown";
   },
@@ -1618,6 +1633,9 @@ export var UrlbarUtils = {
         if (result.payload.keyword === lazy.UrlbarTokenizer.RESTRICT.ACTION) {
           return "restrict_keyword_actions";
         }
+         if (result.payload.keyword === lazy.UrlbarTokenizer.RESTRICT.WORKSPACE) {
+          return "restrict_keyword_workspace";
+        }
     }
 
     return "unknown";
